igvm/vm.py

```diff
@@ -500,3 +500,16 @@ def unblock_autostart(self):
 
     def copy_postboot_script(self, script):
         self.put('/buildvm-postboot', script, '0755')
+
+    def set_safe_blocksize(self):
+        """ Reduce maximum number of KiB allowed for FS to request from
+            block layer
+
+        This is required for DRBD storage migrations. During disk migration
+        a DRBD will consist of devices on multiple hypervisors which might have
+        different maximum block size. It was observed that during high IO
+        requests were sometimes rejected by block layer causing filesystem to
+        crash.
+        """
+        self.run('echo 64 > /sys/class/block/vda/queue/max_sectors_kb')
```

Comment:
can we not query correct block size from the hvs taking part in the migration?

Reply:
From all the information I have I can't really follow how you even conclusively found out it's related to max_sector_kb. Maybe the write just fails at the moment we switch to the drbd device?

Reply:
It keeps failing, so it does not recover from an error until the server is killed.